src/ekat/util/ekat_lin_interp_impl.hpp

```diff
@@ -28,17 +28,17 @@ void LinInterp<ScalarT, PackSize, DeviceT>::setup(const MemberType& team,
   setup_impl(team, *this, ekat::repack<Pack::n>(x1), ekat::repack<Pack::n>(x2));
 }
 
-// Linearly interpolate y(x1) onto coordinates x2
 template <typename ScalarT, int PackSize, typename DeviceT>
 template <typename V1, typename V2, typename V3, typename V4>
 KOKKOS_INLINE_FUNCTION
 void LinInterp<ScalarT, PackSize, DeviceT>::lin_interp(const MemberType& team,
-                                             const V1& x1,
-                                             const V2& x2,
-                                             const V3& y1,
-                                             const V4& y2) const
+                                                       const V1& x1,
+                                                       const V2& x2,
+                                                       const V3& y1,
+                                                       const V4& y2) const
 {
   lin_interp_impl(team,
+                  Kokkos::TeamThreadRange(team, m_km2_pack),
                   *this,
```

Comment:
This is not something that was done in this PR, but I can't help asking: what's the point of passing `*this`? The `lin_interp_impl` routine is probably not meant to be called by the user (possibly passing another LinInterp object). And what would be the use case of doing the following?
```
LinInterp li_1(..);
LinInterp li_2(...);
li_1.lin_interp_impl(team, li_2,...);
```
Unless I'm overlooking something, can we remove the `*this` parameter?

Reply:
@bartgol `liv` is needed in `lin_interp_impl` to provide the `m_indx_map` and other members which are used in the implementation. Calling a static function with a `*this` argument is a common technique I've used when writing classes/structs that launch Kokkos kernels. It's been a while since I've looked at the details, but I think Kokkos doesn't handle the implicit `this` pointer well.

Reply:
`lin_interp_impl` is not a static method though, so you don't need to pass `*this`.

Re: kokkos and *this. The issue is when dispatching a cuda kernel (meaning the outer-most one) using a lambda in the parallel_for. That's b/c the automatic vars need to be copied to device, and `this` is copied instead of `*this` (an issue solved with cuda 11, IIRC). Here, neither concern hold: we're already on device (we are inside a || region), and we are using a member function instead of a lambda in the parallel for).

Reply:
@bartgol , it is: `static void lin_interp_impl`.

Thanks for clarifying the `this` situation. I will see if a regular member function works as the implementation.

Reply:
> @bartgol , it is: `static void lin_interp_impl`.

Ah, you're right, sorry. I was looking at the `_impl.hpp` file, not at the decl one.

Imho, we can do away with static, and remove the extra parameter `LinInterp&`. Since there's no need for a static fcn, emulating a non-static fcn via a static fcn is probably confusing to most people (it is to me, at least).